<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>545</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#MyFrame {background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(100, 0, 50, 255), stop:.9 rgba(255, 255, 255, 255));} </string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="WidgetFrameOuter">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="class" stdset="0">
      <string>PeripheralCollection</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="MyFrame">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="spacing">
          <number>1</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QFrame" name="frame_6">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">#frame_6{background-color:rgb(238, 238, 236)}</string>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <property name="class" stdset="0">
            <string>PeripheralCollectionInner</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <property name="leftMargin">
             <number>2</number>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="rightMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <property name="horizontalSpacing">
             <number>2</number>
            </property>
            <item row="1" column="0" rowspan="3" colspan="3">
             <widget class="QScrollArea" name="scrollArea">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>793</width>
                 <height>535</height>
                </rect>
               </property>
               <layout class="QGridLayout" name="gridLayout_6">
                <property name="leftMargin">
                 <number>2</number>
                </property>
                <property name="topMargin">
                 <number>2</number>
                </property>
                <property name="rightMargin">
                 <number>2</number>
                </property>
                <property name="bottomMargin">
                 <number>2</number>
                </property>
                <property name="spacing">
                 <number>2</number>
                </property>
                <item row="8" column="0">
                 <widget class="QPushButton" name="pushButton_2">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>SINE FIT</string>
                  </property>
                 </widget>
                </item>
                <item row="10" column="0">
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="13" column="0">
                 <widget class="QPushButton" name="pushButton">
                  <property name="text">
                   <string>I/O Controls</string>
                  </property>
                 </widget>
                </item>
                <item row="7" column="0">
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="3" column="0">
                 <widget class="QComboBox" name="optionBox">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="currentIndex">
                   <number>1</number>
                  </property>
                  <item>
                   <property name="text">
                    <string>SR04</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>VL53L0X</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item row="13" column="2">
                 <widget class="QLabel" name="msgwin">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>9</pointsize>
                   </font>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">QLabel{color:red}</string>
                  </property>
                  <property name="text">
                   <string>.</string>
                  </property>
                 </widget>
                </item>
                <item row="9" column="0">
                 <widget class="QPushButton" name="pushButton_3">
                  <property name="text">
                   <string>DAMPED SINE FIT</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="0">
                 <widget class="QCheckBox" name="checkBox_4">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>150</width>
                    <height>25</height>
                   </size>
                  </property>
                  <property name="styleSheet">
                   <string notr="true">border : 2px solid green;</string>
                  </property>
                  <property name="text">
                   <string>Record Data</string>
                  </property>
                 </widget>
                </item>
                <item row="11" column="0">
                 <widget class="QFrame" name="frame">
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>150</height>
                   </size>
                  </property>
                  <property name="frameShape">
                   <enum>QFrame::StyledPanel</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QVBoxLayout" name="gaugeLayout">
                   <property name="spacing">
                    <number>2</number>
                   </property>
                   <property name="leftMargin">
                    <number>2</number>
                   </property>
                   <property name="topMargin">
                    <number>2</number>
                   </property>
                   <property name="rightMargin">
                    <number>2</number>
                   </property>
                   <property name="bottomMargin">
                    <number>2</number>
                   </property>
                  </layout>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QSpinBox" name="durationBox">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="suffix">
                   <string> S</string>
                  </property>
                  <property name="prefix">
                   <string>Duration </string>
                  </property>
                  <property name="minimum">
                   <number>2</number>
                  </property>
                  <property name="maximum">
                   <number>300</number>
                  </property>
                  <property name="value">
                   <number>5</number>
                  </property>
                 </widget>
                </item>
                <item row="12" column="0">
                 <widget class="QPushButton" name="pushButton_6">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>150</width>
                    <height>25</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Save Data</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="2" rowspan="10">
                 <widget class="PlotWidget" name="graph"/>
                </item>
                <item row="5" column="0">
                 <widget class="QSpinBox" name="intervalBox">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="suffix">
                   <string> mS</string>
                  </property>
                  <property name="prefix">
                   <string>Interval </string>
                  </property>
                  <property name="minimum">
                   <number>2</number>
                  </property>
                  <property name="maximum">
                   <number>300</number>
                  </property>
                  <property name="value">
                   <number>2</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>saveTraces()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>7</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>sineFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>dampedSineFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>651</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>durationBox</sender>
   <signal>editingFinished()</signal>
   <receiver>Form</receiver>
   <slot>setDuration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>7</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>show_voltmeter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>537</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optionBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>Form</receiver>
   <slot>setInput(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>-12</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_4</sender>
   <signal>clicked(bool)</signal>
   <receiver>Form</receiver>
   <slot>startLogging(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>-13</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>intervalBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>setInterval(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>-12</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggled()</slot>
  <slot>sineFit()</slot>
  <slot>dampedSineFit()</slot>
  <slot>setDuration()</slot>
  <slot>saveTraces()</slot>
  <slot>show_voltmeter()</slot>
  <slot>setInput(QString)</slot>
  <slot>startLogging(bool)</slot>
  <slot>setInterval(int)</slot>
 </slots>
</ui>
